strawberryfields/gbs/sample.py

```diff
@@ -335,7 +134,7 @@ def to_subgraphs(graph: nx.Graph, samples: list) -> list:
     graph_nodes = list(graph.nodes)
     node_number = len(graph_nodes)
 
-    subgraph_samples = [modes_from_counts(s) for s in samples]
+    subgraph_samples = [list(set(modes_from_counts(s))) for s in samples]
```

Comment:
Why is using list(set( ...)) necessary here?

Reply:
So `modes_from_counts` doesn't guarantee that that output doesn't have repetitions, e.g.:
`modes_from_counts([1, 0, 2, 1]) = [0, 2, 2, 3]`
However, `list(set(...))` will remove repetitions to make a subgraph:
`list(set([0, 2, 2, 3])) = [0, 2, 3]`

The idea here is that, even if the user has generated samples with PNRs, we apply an implicit thresholding here and they can generate subgraphs in the usual way.